corehq/apps/es/management/commands/elastic_sync_multiplexed.py

```diff
@@ -0,0 +1,143 @@
+from datetime import datetime, timedelta
+
+from django.core.management.base import BaseCommand, CommandError
+
+from corehq.apps.es.client import ElasticMultiplexAdapter, get_client
+from corehq.apps.es.client import manager as es_manager
+from corehq.apps.es.exceptions import IndexNotMultiplexedException, TaskMissing
+from corehq.apps.es.registry import get_registry, registry_entry
+from corehq.apps.es.transient_util import doc_adapter_from_info
+from corehq.apps.es.utils import check_task_progress
+
+
+class ESSyncUtil:
+
+    """
+    This class contains methods to support the reindex process of multiplexed indices in HQ
+    """
+
+    def __init__(self):
+        self.es = get_client()
+
+    def start_reindex(self, cname):
+
+        self.adapter = self.get_adapter(cname)
+
+        if not isinstance(self.adapter, ElasticMultiplexAdapter):
+            raise IndexNotMultiplexedException("""Index not multiplexed!
+            Sync can only be run on multiplexed indices""")
+
+        self.source, self.dest = self.get_source_destination_indexes(self.adapter)
+
+        task_id = es_manager.reindex(self.source, self.dest)
+
+        # This would display progress untill reindex process is completed
+        check_task_progress(task_id)
+
+        self.perform_cleanup()
```

Comment:
it the command fails prior to completion is there a way to perform this cleanup?

Reply:
No there is no way to do it at present. I thought about it whether to expose this method or not. The rationale for not exposing it was if reindex fails we would have to run it again which would perform the cleanup anyways.